postgrest/postgrest-postgis-spatial-api.md

```diff
@@ -0,0 +1,403 @@
+
+If you are willing to run PostgreSQL with Docker we recommend you to use [Kartoza's docker recipe](https://hub.docker.com/r/kartoza/postgis/) which comes directly bundled with PostGIS as an extension.
+To this end we will keep it simple and guide you through this tutorial using this image but please feel free to alternatively install PostgreSQL and PostGIS on your host OS.
+Let's start our docker postgres container on port 5432 (or anything else you prefer).
+
+```sh
+sudo docker run --name "postgrest_tut" -p 5432:5432 -e POSTGRES_MULTIPLE_EXTENSIONS=postgis -d -t kartoza/postgis
+```
+
+We will quickly have to configure postgres to make sure it trusts our connections (this is merely for the tutorial and shouldn't be used in production this way).
+
+```sh
+sudo docker exec -it postgrest_tut bash
+```
+
+Inside the container first of all install an editor, e.g. nano, and then navigate to the folder where the postgres config is living. 
+
+```sh
+apt-get update && apt-get install nano
+
+cd /etc/postgresql/11/main/
+```
+
+In `pg_hba.conf` you will have to make a small edit to the settings under `Database administrative login by Unix domain socket` (should be on line 85) from `peer` to `trust` and restart the Docker container afterwards.
+
+```sh
+sudo docker restart postgrest_tut
+```
+
+Afterwards you should be able to execute the following command which will bring up the `psql` prompt.
+
+```sh
+sudo docker exec -it postgrest_tut psql -U postgres
+```
+
+Finally we will have to enable the PostGIS extension.
+
+```sh
+postgres=# CREATE EXTENSION postgis;
+```
+
+Done! 
+Let's move on.
+
+### Install PostgREST 
+
+To keep it simple, we suggest you follow the installation instructions on [postgrest.org](http://postgrest.org/en/v6.0/tutorials/tut0.html) which will depend on your operating system).
+
+Once everything is installed properly you will be able to simply run postgrest with
+
+```sh
+postgrest
+```
+
+And if everything is working correctly it will print out its version and information about configuration.
+
+
+## Step 1 - Creating our Schema
+
+For this tutorial we don't necessarily require a database because we will be implementing logic in PostgreSQL as functions which will spatially compute information on the fly.
+However we will require a schema, so let's bring up the psql prompt of our Docker container (alternatively `psql -U postgres` if it's running on your host OS).
+You should be able to see the `psql` command prompt.
+
+
+```sh
+sudo docker exec -it tutorial psql -U postgres
+
+psql (9.6.3)
+Type "help" for help.
+
+postgres=#
+```
+
+The first thing to do is to create an arbitrary named schema for the database objects which will be exposed in the API. 
+Execute this and the other SQL statements inside the psql prompt you started.
+
+```sh
+create schema api;
+```
+
+Next we should add a role to use for anonymous web requests. 
+And when a request comes in, PostgREST will switch into this role in the database to run queries.
```

Comment:
> Next make a role to use for anonymous web requests. When a request comes in, PostgREST will switch into this role in the database to run queries.
> 
> It’s a good practice to create a dedicated role for connecting to the database, instead of using the highly privileged postgres role. So we’ll do that, name the role authenticator and also grant him the ability to switch to the web_anon role.

It makes more sense for tables etc but I'd like to follow best practice from postgrest authors.